src/app/views/user/user-settings/user-profile/user-profile.component.ts

```diff
@@ -36,29 +64,89 @@ export class UserProfileComponent {
     };
   }
 
+  getCountries() {
+    return (searchText): Promise<Array<Country>> => {
+      return this.countryProxy.getCountries(searchText);
+    };
+  }
+
   onLanguageSelect(language) {
-    if (!isEmpty(language) && !some(this.user.userLanguages, {language: language})) {
-      let userLanguage = new UserLanguage({proficiency: 1});
+    if (!isEmpty(language) && !some(this.user.userLanguages, { language: language })) {
+      let userLanguage = new UserLanguage({ proficiency: 1 });
       userLanguage.language = language;
 
       this.user.userLanguages.push(userLanguage);
     }
   }
 
+  onNativeLanguageSelect(language) {
+    if (language) {
+      let nativeLanguage = new UserLanguage({ proficiency: 5 });
+      nativeLanguage.language = language;
+
+      let oldNativeLanguage = this.user.getNativeLanguage();
```

Comment:
`let` could be changed to [`const`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) here `oldNativeLanguage` and `nativeLanguage` are never re-assigned.

_edit_: Now that I thought checking for `let` => `const` there are a bunch of places where that can be changed. Its really not the end of the world and there is no need for going back and changing it. Though it can be nice to use as a general code style :)